pkg/server/handler.go

```diff
@@ -534,19 +534,27 @@ func (al *activeListener) OnNewConnection(ctx context.Context, conn api.Connecti
 		return
 	}
 	ac := newActiveConnection(al, conn)
+	// connection may be closed by upstream before added to conn list
+	if conn.State() == api.ConnClosed {
+		return
+	}
+
+	if conn.LocalAddr().Network() == "udp" {
+		network.SetUDPProxyMap(network.GetProxyMapKey(conn.LocalAddr().String(), conn.RemoteAddr().String()), conn)
+	}
 
 	e := al.conns.PushBack(ac)
 	ac.element = e
 
+	// add event listener after setting ac.element to avoid delete empty element
```

Comment:
Do you mean panic here?
https://github.com/mosn/mosn/blob/aab8c094446abf98652b6b68a798ba553dff0dad/pkg/server/handler.go#L640

Reply:
Yes, that's what I mean. If event listener added before ```ac.element``` set, ``` al.conns.Remove(ac.element) ``` can be called ealier

Reply:
got it, thanks!